File: 프로그래머스/3/152995. 인사고과/인사고과.js
function solution(scores) {
    let answer = 1;
    let [wanhoAt, wanhoEx] = scores[0];
    scores.sort((a, b) => b[0] - a[0] || a[1] - b[1]);

    let passNum = 0;
    
    for (let i = 0; i < scores.length; i++) {
        let [checkAt, checkEx] = scores[i];

        if (wanhoAt < checkAt && wanhoEx < checkEx) {
            return -1;
        }

        if (passNum <= checkEx) {
            if (wanhoAt + wanhoEx < checkAt + checkEx) {
                answer++;
            }
            passNum = checkEx;
        }
    }

    return answer;
}

File: 프로그래머스/3/152995. 인사고과/인사고과.py
def solution(scores):
    answer = 1
    wanho_at,wanho_ex = scores[0]
    scores.sort(key=lambda x:(-x[0],x[1]))
    pass_num = 0
    for score in scores:
        check_at, check_ex = score
        if wanho_at < check_at and wanho_ex < check_ex:
            return -1
        
        if pass_num <= check_ex:
            if wanho_at+wanho_ex < check_at+check_ex:
                answer+=1
            pass_num = check_ex
    return answer